### DISCUSSION TITLE
RTKQ: selectFromResult is not applied to lazy hook trigger result

### DISCUSSION THREAD

**Discussion by borispinus** [reduxjs/redux-toolkit#3938] [Q&A] [answered]
https://github.com/reduxjs/redux-toolkit/blob/cf2441945d29c0fe263a9cffa103f6b33ebcd7b3/packages/toolkit/src/query/react/buildHooks.ts#L812

According to this piece of code `useLazyQuerySubscription` ignores `selectFromResult` option (which is passed in `useLazyQuery`). Because of that `trigger().unwrap()` returns original(not mapped data)

Is it expected behaviour or not?

**Comment by EskiMojo14** [comment, marked as answer]
this is intentional - selectFromResult is for the hook result, as it decides when the component rerenders 

the trigger on the other hand can return whatever without affecting the component rerendering, and you can call any selectors you want on its result
